vendor/bat-native-ads/src/bat/ads/internal/account/account.cc

```diff
@@ -95,24 +96,26 @@ void Account::MaybeGetIssuers() const {
   issuers_->MaybeFetch();
 }
 
-void Account::DepositFunds(const std::string& creative_instance_id,
-                           const AdType& ad_type,
-                           const ConfirmationType& confirmation_type) {
+void Account::Deposit(const std::string& creative_instance_id,
+                      const AdType& ad_type,
+                      const ConfirmationType& confirmation_type) {
   DCHECK(!creative_instance_id.empty());
   DCHECK_NE(AdType::kUndefined, ad_type.value());
   DCHECK_NE(ConfirmationType::kUndefined, confirmation_type.value());
 
-  database::table::CreativeAds database_table;
-  database_table.GetForCreativeInstanceId(
-      creative_instance_id,
-      [=](const bool success, const std::string& creative_instance_id,
-          const CreativeAdInfo& creative_ad) {
+  std::unique_ptr<DepositBase> deposit =
+      DepositsFactory::Build(confirmation_type);
+  DCHECK(deposit);
+
+  deposit->GetValue(
+      creative_instance_id, [=](const bool success, const double value) {
         if (!success) {
-          NotifyFailedToDepositFunds(creative_ad, ad_type, confirmation_type);
+          NotifyFailedToProcessDeposit(creative_instance_id, ad_type,
+                                       confirmation_type);
           return;
         }
 
-        Credit(creative_ad, ad_type, confirmation_type);
+        ProcessDeposit(creative_instance_id, ad_type, confirmation_type, value);
```

Comment:
Adds the transaction to the transactions database table and sends confirmation to the server